lib/sass/stack.rb

```diff
@@ -98,6 +98,16 @@ def with_mixin(filename, line, name)
       with_frame(filename, line, :mixin, name) {yield}
     end
 
+    # Pushes a function frame onto the stack.
+    #
+    # @param filename [String] See \{Frame#filename}.
+    # @param line [String] See \{Frame#line}.
+    # @param name [String] See \{Frame#name}.
+    # @yield [] A block in which the new frame is on the stack.
+    def with_function(filename, line, name)
+      with_frame(filename, line, :function, name) {yield}
+    end
```

Comment:
I also needed to add a frame for @warn to get the trace to output the previous caller (the function).